pkgs/development/libraries/libcaca/default.nix

```diff
@@ -13,8 +15,16 @@ stdenv.mkDerivation rec {
 
   outputs = [ "bin" "dev" "out" "man" ];
 
-  propagatedBuildInputs = [ ncurses zlib imlib2 pkgconfig libX11 ]
-   ++ stdenv.lib.optional stdenv.isDarwin libXext;
+  configureFlags = [
+    (if x11Support then "--enable-x11" else "--disable-x11")
+    ];
+
+  NIX_CFLAGS_COMPILE = stdenv.lib.optional (!x11Support) "-DX_DISPLAY_MISSING";
+
+  enableParallelBuilding = true;
+
+  propagatedBuildInputs = [ ncurses zlib imlib2-nox pkgconfig ]
```

Comment:
I see that imlib2-nox explicitly disables X11 support. Wouldn't this be better as:
`propagatedBuildInputs = [ ncurses zlib pkgconfig ] ++ stdenv.lib.optionals x11Support [ imlib2 libX11 libxext ] ++ stdenv.lib.optional imlib2-nox` or better yet `propagatedBuildInputs = [ ncurses zlib pkgconfig (imlib2.override { inherit x11Support; }) ] ++ stdenv.lib.optionals x11Support [ libX11 libxext ];`?

Reply:
Done.

Reply:
Awesome thanks!